plugins/googleanalytics/src/org/pentaho/di/ui/trans/steps/googleanalytics/GaInputStepDialog.java

```diff
@@ -971,8 +972,16 @@ public void handleEvent( Event e ) {
             wFields.optWidth( true );
             input.setChanged();
           } catch ( IOException ioe ) {
+            Exception exceptionToDisplay = ioe;
+            // Try to display something more user friendly than plain JSON
+            if ( ioe instanceof GoogleJsonResponseException ) {
+              GoogleJsonResponseException gjre = (GoogleJsonResponseException) ioe;
+              if ( gjre.getDetails() != null && gjre.getDetails().getMessage() != null ) {
+                exceptionToDisplay = new RuntimeException( gjre.getDetails().getMessage(), gjre );
```

Comment:
That was my initial idea, but I disregarded it in favour of just generic RuntimeException caused by the original one.